json.Unmarshaler interface not considered for nested types

I believe I may have found a false positive. The following minimum reproduction shows the false positive:

```
package main

import (
    "encoding/json"
    "fmt"
)

type Outer struct {
    Inner Inner `json:"inner"`
}

type Inner struct {
    String string
}

func (i *Inner) UnmarshalJSON(p []byte) error {
    var s string
    if err := json.Unmarshal(p, &s); err != nil {
        return err
    }
    i.String = s
    return nil
}

func main() {
    buf := []byte(`
        {
            "inner": "my string"
        }
    `)
    var outer Outer
    if err := json.Unmarshal(buf, &outer); err != nil {
        panic(err)
    }
    fmt.Println(outer)
}
```

The output of the program is:

```
{{my string}}
```

See https://go.dev/play/p/HEcyBYmfO_Y

However, `musttag` with the latest tag (v0.12.1) shows a false positive:

```
main.go:32:32: the given struct  should be annotated with the `json` tag
```

It doesn't seem to ignore the absence of JSON tags on the `Inner` type, even though the `Inner` type implements the `jsonUnmarshaler` interface.

--- a/musttag.go
+++ b/musttag.go
@@ -273,7 +273,7 @@ func implementsInterface(typ types.Type, ifaces []string, imports []*types.Packa
 		if !ok {
 			continue
 		}
-		if types.Implements(typ, iface) {
+		if types.Implements(typ, iface) || types.Implements(types.NewPointer(typ), iface) {
 			return true
 		}
 	}

--- a/testdata/src/tests/tests.go
+++ b/testdata/src/tests/tests.go
@@ -131,23 +131,17 @@ func shouldBeIgnored() {
 	json.Marshal(nil) // nil argument, see issue #20.
 }
 
-type WithInterface struct {
-	NoTag string
-}
-
-func (w WithInterface) MarshalJSON() ([]byte, error) {
-	return json.Marshal(w.NoTag)
-}
-
 func nestedTypeWithInterface() {
 	type Foo struct {
-		Nested WithInterface `json:"nested"`
+		Nested Marshaler `json:"nested"`
 	}
 	var foo Foo
-	json.Marshal(foo)    // no error
-	json.Marshal(&foo)   // no error
-	json.Marshal(Foo{})  // no error
-	json.Marshal(&Foo{}) // no error
+	json.Marshal(foo)
+	json.Marshal(&foo)
+	json.Marshal(Foo{})
+	json.Marshal(&Foo{})
+	json.Unmarshal(nil, &foo)
+	json.Unmarshal(nil, &Foo{})
 }
 
 func ignoredNestedType() {
@@ -159,10 +153,10 @@ func ignoredNestedType() {
 		Exported string `json:"exported"`
 	}
 	var foo Foo
-	json.Marshal(foo)    // no error
-	json.Marshal(&foo)   // no error
-	json.Marshal(Foo{})  // no error
-	json.Marshal(&Foo{}) // no error
+	json.Marshal(foo)
+	json.Marshal(&foo)
+	json.Marshal(Foo{})
+	json.Marshal(&Foo{})
 }
 
 func ignoredNestedTypeWithSubsequentNoTagField() {

